purchases/src/main/kotlin/com/revenuecat/purchases/utils/OfferingImagePreDownloader.kt

```diff
@@ -27,14 +47,140 @@ internal class OfferingImagePreDownloader(
 
         debugLog("OfferingImagePreDownloader: starting image download")
 
+        downloadV1Images(offering)
+        downloadV2Images(offering)
+    }
+
+    private fun downloadV1Images(offering: Offering) {
         offering.paywall?.let { paywallData ->
             val imageUris = paywallData.config.images.all.map {
                 Uri.parse(paywallData.assetBaseURL.toString()).buildUpon().path(it).build()
             }
             imageUris.forEach {
-                debugLog("Pre-downloading paywall image: $it")
+                debugLog("Pre-downloading Paywall V1 paywall image: $it")
+                coilImageDownloader.downloadImage(it)
+            }
+        }
+    }
+
+    private fun downloadV2Images(offering: Offering) {
+        offering.paywallComponents?.let { paywallComponents ->
+            val imageUrls = findImageUrisToDownload(paywallComponents)
+            imageUrls.forEach {
+                debugLog("Pre-downloading Paywall V2 paywall image: $it")
                 coilImageDownloader.downloadImage(it)
             }
         }
     }
+
+    private fun findImageUrisToDownload(paywallComponents: Offering.PaywallComponents): Set<Uri> {
+        val paywallComponentsConfig = paywallComponents.data.componentsConfig.base
+
+        return paywallComponentsConfig.stack.findImageUrisToDownload() +
+            (paywallComponentsConfig.stickyFooter?.stack?.findImageUrisToDownload() ?: emptyList()) +
```

Comment:
```suggestion
            paywallComponentsConfig.stickyFooter?.stack?.findImageUrisToDownload().orEmpty() +
```

(Don't apply on GitHub, probably needs an import 😅)